File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    let x_num = 0, cons = 0;
    
    polynomial.split(' + ').forEach(i => {
        if (Number(i)) {
            cons += Number(i);
        } else {
            x_num += i == 'x' ? 1 : Number(i.slice(0, -1));
        }
    })
    
    if (x_num == 0) {
        return String(cons);
    } else if (x_num == 1) {
        return cons == 0 ? 'x' : ('x + ' + cons);
    } else {
        return cons == 0 ? (x_num + 'x') : (x_num + 'x + ' + cons);
    }
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x_num, const = 0, 0
    
    for i in polynomial.split(' + '):
        if i.isdigit():
            const += int(i)
        else:
            x_num += 1 if i == 'x' else int(i[:-1])
            
    if x_num == 0:
        return f"{const}"
    elif x_num == 1:
        return 'x' if const == 0 else f"x + {const}"
    else:
        return f"{x_num}x" if const == 0 else f"{x_num}x + {const}"